### ISSUE TITLE
[Question] @ResolveProperty Options

### ISSUE THREAD

**Issue by johannesschobel** [nestjs/graphql#451]
## I'm submitting a...
<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
I am using NestJS with the `Code First` Approach (TypeGraphQL) as described in the official docs of NestJS.

In the `@ResolveProperty()` decorator there is the possibility to add custom options - for example, a `deprecationMessage`, `nullable` or a `description`.

See the following Code Snippet:
```ts
@ResolveProperty(type => Date, {name: "createdAt"})
async resolveCreatedAt(@Parent() parent: User): Promise<Date> {
   return await new Date(parent.registerDate);
}
```

So basically, i would like to "rename" my internal `registerDate` and output it as `createdAt` name. I thought, that the `AdvancedOptions.name` field may be used for this. However, this does not work.

In fact, it raises an error:
```
(node:24828) UnhandledPromiseRejectionWarning: Error: User.resolveCreatedAt defined in resolvers, but not in schema
(node:24828) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:24828) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I am aware, that my `User` (`@ObjectType`) does not have a `@Field() resolveCreatedAt`, however it does have a `@Field() createdAt`.

I had to do it as follows:
```ts
@ResolveProperty("createdAt", type => Date)
async resolveCreatedAt(...) { ... }
```

What does the `name` property actually do? Why is it used / for what is it used?

All the best and thank you very much!

## Expected behavior

I thought the `AdvancedOptions.name` would allow me to "rename" the property.

## Environment

<pre><code>
Nest Core: 6.7.2
Nest Common: 6.7.2
Nest GraphQL: 6.5.3
</code></pre>

**Comment by kamilmysliwiec** [comment]
Please, provide a minimal repository which reproduces your issue.

**Comment by johannesschobel** [comment]
Dear @kamilmysliwiec ,

thank you very much for your reply. I just created a demo repository to illustrate this issue: https://github.com/johannesschobel/nestjs-graphql-issue-451

Just run
```
npm install
npm run start
```

to boot the failing application. This repository is based on your `samples/23-type-graphql` example.

Note that i have 2 `@ResolveProperty()` methods in the `recipe.resolver.ts` file, that should "transform" respective input.

Please note that the application **does not boot up** properly, because of one of the failing `@ResolveProperty()` methods (see issue described above).

If you require any more information / assistance, please feel free to contact me!
You do a really great job on this framework, i absolutely love it and i appreciate your hard work in this framework!

All the best

**Comment by kamilmysliwiec** [comment]
Have you tried `@ResolveProperty('title', type => String)` instead?

**Comment by johannesschobel** [comment]
Dear @kamilmysliwiec ,
yeah, i did it like this
```
@ResolveProperty("createdAt", type => Date)
async resolveCreatedAt(...) { ... }
```

however, i am wondering, what the `AdvancedOptions.name` property does - i thought, that this `name` may be used for "mapping".. Can you please clarify this?

All the best

**Comment by kamilmysliwiec** [comment]
It should be fixed in 6.5.4 :) Let me know if you face any issues
